doc/storage.md

```diff
@@ -350,6 +383,38 @@ sudo btrfs filesystem resize max /var/lib/lxd/storage-pools/<POOL>/
    settings in `/etc/lvm/lvm.conf` to avoid slowdowns when interacting with
    LXD.
 
+#### Storage pool configuration
+Key                             | Type      | Condition                         | Default                    | Description
+:--                             | :---      | :--------                         | :------                    | :----------
+source                          | string    | -                                 | -                          | Path to block device or loop file or filesystem entry
+rsync.bwlimit                   | string    | -                                 | 0 (no limit)               | Specifies the upper limit to be placed on the socket I/O whenever rsync has to be used to transfer storage entities.
+rsync.compression               | bool      | appropriate driver                | true                       | Whether to use compression while migrating storage pools.
+volatile.initial\_source        | string    | -                                 | -                          | Records the actual source passed during creating (e.g. /dev/sdb).
+volatile.pool.pristine          | string    | -                                 | true                       | Whether the pool has been empty on creation time.
+volume.size                     | string    | appropriate driver                | unlimited (10GiB for block)| Default volume size
+lvm.thinpool\_name              | string    | lvm driver                        | LXDThinPool                | Thin pool where volumes are created.
+lvm.use\_thinpool               | bool      | lvm driver                        | true                       | Whether the storage pool uses a thinpool for logical volumes.
+lvm.vg\_name                    | string    | lvm driver                        | name of the pool           | Name of the volume group to create.
+lvm.vg.force\_reuse             | bool      | lvm driver                        | false                      | Force using an existing non-empty volume group.
+volume.lvm.stripes              | string    | lvm driver                        | -                          | Number of stripes to use for new volumes (or thin pool volume).
+volume.lvm.stripes.size         | string    | lvm driver                        | -                          | Size of stripes to use (at least 4096 bytes and multiple of 512bytes).
+volume.block.filesystem         | string    | block based driver (lvm)          | ext4                       | Filesystem to use for new volumes
+volume.block.mount\_options     | string    | block based driver (lvm)          | discard                    | Mount options for block devices
```

Comment:
So there is an inconsistency with those across the different drivers.
Basically you can set `volume.KEY` on the pool for any valid volume key defined below for the given driver.

So we either need to make sure to duplicate the entire set of volume keys into the pool config with a `volume.` prefix, or we should just drop them from all pool configs and make sure we have a dedicated section in the doc which explains the pool to volume relationship and that you can use `volume.KEY` on the pool for any valid volume key so you can set its default within that pool.